go/bin/datadog/main.go

```diff
@@ -1,60 +1,65 @@
 package main
 
 import (
-	"log"
-	"os"
-	"time"
-
-	"github.com/dylibso/observe-sdk/go"
-	"github.com/dylibso/observe-sdk/go/adapter/datadog"
-	"github.com/tetratelabs/wazero"
-	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
+  "context"
+  "log"
+  "os"
+  "time"
+
+  "github.com/dylibso/observe-sdk/go/adapter/datadog"
+  "github.com/tetratelabs/wazero"
+  "github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
 )
 
 func main() {
-	//
-	// Collector API
-	collector := observe.NewCollector(nil)
-	ctx, r, err := collector.InitRuntime()
-	if err != nil {
-		log.Panicln(err)
-	}
-	defer r.Close(ctx) // This closes everything this Runtime created.
-
-	// Load WASM from disk
-	wasm, err := os.ReadFile(os.Args[1])
-	if err != nil {
-		log.Panicln(err)
-	}
-
-	// Wazero stuff
-	// Instantiate WASI
-	wasi_snapshot_preview1.MustInstantiate(ctx, r)
-	// collector.CustomEvent("Start module", map[string]interface{}{"name": "testing"})
-
-	//
-	// Adapter API
-	adapter, err := datadog.NewDatadogAdapter(nil)
-	if err != nil {
-		log.Panicln(err)
-	}
-	adapter.Start(collector, wasm)
-	defer adapter.Wait(collector, time.Millisecond*100)
-	defer adapter.Stop(collector)
-
-	adapter.SetTraceId(datadog.NewTraceId())
-
-	config := wazero.NewModuleConfig().
-		WithStdin(os.Stdin).
-		WithStdout(os.Stdout).
-		WithStderr(os.Stderr).
-		WithArgs(os.Args[1:]...).
-		WithStartFunctions("_start")
-	m, err := r.InstantiateWithConfig(ctx, wasm, config)
-	if err != nil {
-		log.Panicln(err)
-	}
-	defer m.Close(ctx)
-
-	time.Sleep(time.Second * 2)
+  ctx := context.Background()
+
+  log.Println("Starting adapter")
+
+  // we only need to create and start once per instance of our host app
+  ddconf := datadog.DefaultDatadogConfig()
+  adapter, err := datadog.NewDatadogAdapter(ddconf)
+  adapter.Start()
+  defer adapter.Stop()
+
+  log.Println("Adapter started")
+
+  // Load WASM from disk
+  wasm, err := os.ReadFile(os.Args[1])
+  if err != nil {
+    log.Panicln(err)
+  }
+
+  log.Println("Create trace ctx")
+  traceCtx, err := adapter.NewTraceCtx(wasm, nil)
+  if err != nil {
+    log.Panicln(err)
+  }
+  log.Println("trace ctx created")
+  cfg := wazero.NewRuntimeConfig().WithCustomSections(true)
+  rt := wazero.NewRuntimeWithConfig(ctx, cfg)
+  err = traceCtx.Init(ctx, rt)
```

Comment:
What do you think about doing `NewTraceCtx` + `Init` all in one function? this way the `NewTraceCtx` could take in the `ctx` + `rt` and not have to call `traceCtx.Init` at all? haven't looked yet at the Init impl though so there could be some requirement

Reply:
IMO the fewer steps we need to fully integrate the better

Reply:
I think the problem was around what needed to be on the context. let me try to shorten it. we still want them to explicitly instantiate the runtime though right? we could go back to hiding it and just have them pass in the runtimeconfig

Reply:
does this seem better? it seems to work:

```go
  cfg := wazero.NewRuntimeConfig().WithCustomSections(true)
  rt := wazero.NewRuntimeWithConfig(ctx, cfg)
  traceCtx, err := adapter.NewTraceCtx(ctx, rt, wasm, nil)
  if err != nil {
    log.Panicln(err)
  }
  wasi_snapshot_preview1.MustInstantiate(ctx, rt)
```